### PR TITLE
Move storage to new BlobSidecar

### PR DESCRIPTION
**Pull request by zilm13**
## PR Description
Part of #7654 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: networking/eth2/src/integration-test/java/tech/pegasys/teku/networking/eth2/BlobSidecarsByRangeIntegrationTest.java [hunk 1]
```diff
@@ -113,29 +108,22 @@ public void requestBlobSidecars_shouldReturnCanonicalBlobSidecarsOnDenebMileston
...
   }
 
-  private List<BlobSidecarOld> requestBlobSidecarsByRange(
+  private List<BlobSidecar> requestBlobSidecarsByRange(
       final Eth2Peer peer, final UInt64 from, final UInt64 count)
       throws InterruptedException, ExecutionException, TimeoutException {
     final List<BlobSidecar> blobSidecars = new ArrayList<>();
     waitFor(
         peer.requestBlobSidecarsByRange(from, count, RpcResponseListener.from(blobSidecars::add)));
-    waitFor(() -> assertThat(peer.getOutstandingRequests()).isEqualTo(0));
```
Comment (tbenr) [inline networking/eth2/src/integration-test/java/tech/pegasys/teku/networking/eth2/BlobSidecarsByRangeIntegrationTest.java:133]: is it ok to remove this `waitFor`?

**Review by tbenr — APPROVED** [review]
LGTM. just one thing

**Comment by zilm13** [comment]
@tbenr great catch. Accidentally removed with repack removing, checking other places now.

**Comment by zilm13** [comment]
Looks good now, merging

**Pull request merged**